### PR TITLE
Allow GCS client to use credentials from the environment if needed

### PR DESCRIPTION
**Pull request by cmaster11**
When running in GCP environment, the credentials can be fetched automatically by the GCP SDK, and there is no need to specify any additional JSON/env variables (https://cloud.google.com/docs/authentication/production#automatically).

### CODE REVIEW

**Comment by Xuanwo** [comment]
diff check failed, use `make build` can resolve this problem.

File: utils.go [hunk 1]
```diff
@@ -111,20 +109,25 @@ func newServicer(pairs ...typ.Pair) (srv *Service, err error) {
 		if err != nil {
 			return nil, err
 		}
+	case credential.ProtocolEnv:
```
Comment (Xuanwo) [inline utils.go:112]: We need to support both env and other ways.

How about using `google.FindDefaultCredentials` instead?


ref: https://pkg.go.dev/golang.org/x/oauth2/google#FindDefaultCredentials

Comment (cmaster11) [reply utils.go:112]: `FindDefaultCredentials` gets called automatically by the GCS client if no credentials are provided (e.g. if you don't use the `WithCredentialsJSON`).

So, if you pass `env` as credential type, the client will just fetch them all at runtime. 

This happens here: https://github.com/googleapis/google-api-go-client/blob/master/internal/creds.go#L50

**Comment by cmaster11** [comment]
> diff check failed, use `make build` can resolve this problem.

Ran and diff is fine now :)

File: utils.go (hunk 1 above)
Comment (Xuanwo) [reply utils.go:112]: > gets called automatically by the GCS client if no credentials are provided

Yep, I know about this logic here.

But I don't want to depend on this behavior. I prefer to control them by ourselves.

- If the user uses `base64` or `file`, it's OK, read them and use `WithCredentialsJSON` to construct the client.
- If the user uses `env`, we use `FindDefaultCredentials` to read default credentials from the env.

Everything is clear, no need to read the code provided by deps.

What do you think?

Comment (cmaster11) [reply utils.go:112]: Absolutely got your point :) I cleaned up the code

**Review by Xuanwo — APPROVED** [review]

**Comment by Xuanwo** [comment]
Thanks for your contributions!

**Pull request merged**